src/lib/addons/slack.ts

```diff
@@ -10,6 +10,13 @@ import {
 } from './feature-event-formatter-md';
 import { IEvent } from '../types/events';
 
+interface ISlackAddonParameters {
+    url: string;
+    username?: string;
+    defaultChannel: string;
+    emojiIcon?: string;
+    customHeaders?: string;
```

Comment:
Maybe this would benefit from the same suggestion:
```
customHeaders?: { [key: string]: string } | Record<string, string> ;
```

Reply:
Same comment here, since we're doing a post of parameters, it isn't currently possible to have it as anything but strings, otherwise our template for event data would also be something else but string. I tested locally to ensure that this was correct.